File: csc345/homework_0/References.java
public class References {
    static class Int {
        int value;

        public Int(int value) {
            this.value = value;
        }

        public void setInt(int value) {
            this.value = value;
        }

        public int getInt() {
            return value;
        }
    }

    private static int square(Int i) {
        i.setInt(i.getInt() * i.getInt());

        return i.getInt();
    }

    private static int twice(Int i) {
        i.setInt(2 * i.getInt());

        return i.getInt();
    }

    private static int once(Int i) {
        return i.getInt();
    }

    public static void main(String[] args) {
        Int x = new Int(3);
        int y = square(x) + twice(x) + once(x);
        System.out.println(y);
    }
}

File: csc345/homework_0/references.cpp
#include <iostream>

using namespace std;

int square(int& i);
int twice(int& i);
int once(int& i);

int main() {
    int x = 3;
    int y = square(x) + twice(x) + once(x);
    cout << y << endl;
}

int square(int& i) {
    return i = i * i;
}

int twice(int& i) {
    return i = 2 * i;
}

int once(int& i) {
    return i;
}